blueocean-github-pipeline/src/test/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubOrgFolderPermissionsTest.java

```diff
@@ -0,0 +1,166 @@
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(false);
+    }
+
+    private void createGithubOrgFolder(boolean shouldSuceed) throws Exception {
+        String credentialId = createGithubCredential(user);
+        String orgFolderName = "cloudbeers1";
+        Map resp = new RequestBuilder(baseUrl)
+                .status(shouldSuceed ? 201 : 403)
+                .jwtToken(getJwtToken(j.jenkins,user.getId(), user.getId()))
+                .post("/organizations/" + getOrgName() + "/pipelines/")
+                .data(ImmutableMap.of("name", orgFolderName,
+                        "$class", "io.jenkins.blueocean.blueocean_github_pipeline.GithubPipelineCreateRequest",
+                        "scmConfig", ImmutableMap.of("config",
+                                ImmutableMap.of("repos", ImmutableList.of("PR-demo"), "orgName","cloudbeers"),
+                                "credentialId", credentialId,
+                                "uri", githubApiUrl)
+                ))
+                .build(Map.class);
+
+        TopLevelItem item = getOrgRoot().getItem(orgFolderName);
+        if (shouldSuceed) {
+            assertEquals(orgFolderName, resp.get("name"));
+            assertEquals("io.jenkins.blueocean.blueocean_github_pipeline.GithubOrganizationFolder", resp.get("_class"));
+
+            Assert.assertTrue(item instanceof OrganizationFolder);
+            Map r = get("/organizations/"+ getOrgName() + "/pipelines/"+orgFolderName+"/");
+            assertEquals(orgFolderName, r.get("name"));
+            assertFalse((Boolean) r.get("scanAllRepos"));
+        }
+        else {
+            assertEquals(403, resp.get("code"));
+            assertEquals("Failed to create pipeline: cloudbeers1. User vivek doesn't have Job create permission", resp.get("message"));
+            Assert.assertNull(item);
+            String r = get("/organizations/"+ getOrgName() + "/pipelines/"+orgFolderName+"/", 404, String.class);
+        }
+    }
+
+    private String createGithubCredential(User user) throws UnirestException {
+        Map r = new RequestBuilder(baseUrl)
+                .data(ImmutableMap.of("accessToken", "12345"))
+                .status(200)
+                .jwtToken(getJwtToken(j.jenkins, user.getId(), user.getId()))
+                .put("/organizations/" + getOrgName() + "/scm/github/validate/")
+                .build(Map.class);
+
+        assertEquals("github", r.get("credentialId"));
+        return "github";
+    }
+
+    private static String getOrgName() {
+        return OrganizationFactory.getInstance().list().iterator().next().getName();
+    }
+
+    private static ModifiableTopLevelItemGroup getOrgRoot() {
+        return OrganizationFactory.getItemGroup(getOrgName());
+    }
+
+    @TestExtension(value={"canCreateWhenHavePermissionsOnCustomOrg","canNotCreateWhenHaveNoPermissionOnCustomOrg"})
```

Comment:
Now what I really want is a `TestExtension` where I can set a value on a Class and method so I can use the same `TestExtension` in multiple tests rather than cut & pasting - but well C&P for tests is not so bad (and I can always create a parent and just extend it with the annotation if I was that bothered about code duplication...